joplin/pages/base_page/models.py

```diff
@@ -104,42 +105,20 @@ def janis_preview_url_end(self, revision=None):
             # Janis will query from its default CMS_API if a param is not provided
             return url_end + f"?CMS_API={settings.CMS_API}"
 
-    def janis_url_base(self, janis_branch):
-        """
-        returns a valid url of the base URL in janis:
-            Use hardcoded JANIS_URL for staging and prod
-            Otherwise, use configurable branch setting
-
-        TODO: this and url_base in site settings could probably
-        be revisited for semantics to be less confusing
-        """
-        if settings.ISSTAGING or settings.ISPRODUCTION:
-            return os.getenv("JANIS_URL")
-        else:
-            branch_settings = JanisBranchSettings.objects.first()
-            if branch_settings:
-                return branch_settings.url_base(janis_branch)
-            else:
-                # If we've made it here, we don't have a base url
-                # let's just make up a fake one
-                return "http://fake.base.url/"
```

Comment:
this helped avoid some errors, probably want to do a little testing around what happens with missing settings

Reply:
Hmm. I'll add that back in then. We might need another way of handling errors for this. The problem is that fake.base.url won't work anyway for previewing or publishing.